### ISSUE TITLE
Python 3.8 now throws an exception when modifying a dictionary while iterating

### ISSUE THREAD

**Issue by doko42** [biopython/biopython#2325]
Python 3.8 now throws an exception when modifying a dictionary while iterating

```
--- a/Bio/SubsMat/__init__.py
+++ b/Bio/SubsMat/__init__.py
@@ -207,7 +207,7 @@ class SeqMat(dict):
 
     def _correct_matrix(self):
         """Sort key tuples (PRIVATE)."""
-        for key in self:
+        for key in dict(self):
             if key[0] > key[1]:
                 self[(key[1], key[0])] = self[key]
                 del self[key]
```

**Comment by peterjc** [comment]
Rather than doing ``dict(self)`` probably ``list(self)`` would be enough. Do you have a test case triggering this? Our existing unit tests evidently don't cover it...

**Comment by doko42** [comment]
https://launchpad.net/ubuntu/+source/python-biopython/1.73+dfsg-1ubuntu3
please see the debian/rules file or at the build logs how the tests are invoked. I looks to me like the upstream testsuite, however that's still 1.73.

**Comment by peterjc** [comment]
I don't understand why with our master branch (where nothing has changed in this regard) the same tests pass on Python 3.8, but the issue itself seems clear.

We ere using a pre-release, but here it is Python 3.8.0 (final):
https://travis-ci.org/biopython/biopython/jobs/608217669

Anyway, @mdehoon do you want to take this one?

**Comment by peterjc** [comment]
See this now on the Biopython 1.75 wheel tests (still puzzled why our main TravisCI setup missed it), Linux:

https://travis-ci.org/biopython/biopython-wheels/jobs/608308357
https://travis-ci.org/biopython/biopython-wheels/jobs/608308358

macOS probably too (waiting for that to run).

**Comment by peterjc** [comment]
I understand the test results now, on our main TravisCI:

```
test_SubsMat ... skipping. Install NumPy if you want to use Bio.SubsMat.
```

Until recently, NumPy was not available for Python 3.8 - it is now, and the wheels had it installed for testing.
